test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +225,110 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+	}
+
+	nad, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s: %w", networkName, err)
+	}
+
+	var nadConfig struct {
+		IPAM struct {
+			IPPools []string `json:"ippools"`
+		} `json:"ipam"`
+	}
+
+	if err := json.Unmarshal([]byte(nad.Spec.Config), &nadConfig); err != nil {
+		return nil, fmt.Errorf("failed to parse NAD config JSON: %w", err)
+	}
+
+	return nadConfig.IPAM.IPPools, nil
+}
+
+// checkIPReleased verifies if the VLAN IP is released.
+func (data *testData) checkIPReleased(ipPoolName, podIPString string) error {
+	crdClient, err := data.e2eTestData.GetCRDClient()
+	if err != nil {
+		return fmt.Errorf("failed to get CRD client: %w", err)
+	}
+
+	// Poll every 5 seconds for up to 40 seconds
+	ctx, cancel := context.WithTimeout(context.Background(), 40*time.Second)
+	defer cancel()
+
+	return wait.PollUntilContextTimeout(ctx, 5*time.Second, 40*time.Second, true, func(ctx context.Context) (bool, error) {
+		ipPool, err := crdClient.CrdV1beta1().IPPools().Get(context.TODO(), ipPoolName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName, err)
+		}
+
+		for _, ipAddress := range ipPool.Status.IPAddresses {
+			if podIPString == ipAddress.IPAddress {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+}
+
+// reconcilationAfterAgentRestart verifies OVS cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	vlanPod := data.pods[1]
+	iface := "eth1"
+
+	beforeIPs, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get Pod IP before agent restart: %w", err)
+	}
+
+	beforeIP, exists := beforeIPs[iface]
+	if !exists || beforeIP == nil {
+		return fmt.Errorf("no IP found for interface %s before agent restart", iface)
+	}
+
+	// Restarting the Antrea agent
+	if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+		t.Fatalf("Failed to restart Antrea agent pods: %v", err)
+	}
+
+	afterIPs, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get Pod IP after agent restart: %w", err)
+	}
+
+	afterIP, exists := afterIPs[iface]
+	if !exists || afterIP == nil || !beforeIP.Equal(afterIP) {
+		return fmt.Errorf("OVS port/IP mismatch after agent restart: before=%v, after=%v, iface=%v", beforeIP, afterIP, iface)
+	}
+
+	ipPools, err := data.getIPPoolNames(vlanPod.interfaceNetworks[iface], "default")
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool: %w", err)
+	}
+
+	// Remove Pod and check IP released or not.
+	if err := data.e2eTestData.DeletePodAndWait(defaultTimeout, vlanPod.podName, data.e2eTestData.GetTestNamespace()); err != nil {
+		return fmt.Errorf("failed to delete Pod %s: %w", vlanPod.podName, err)
+	}
+
+	return data.checkIPReleased(ipPools[0], beforeIP.String())
```

Comment:
Could we check if OVS ports are removed?

Reply:
yes refactoring the code, will update the PR.